app/src/main/java/org/wikipedia/readinglist/recommended/RecommendedReadingListViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package org.wikipedia.readinglist.recommended
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import org.wikipedia.Constants
+import org.wikipedia.database.AppDatabase
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.page.PageTitle
+import org.wikipedia.readinglist.database.RecommendedPage
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.Resource
+
+class RecommendedReadingListViewModel : ViewModel() {
+
+    private val handler = CoroutineExceptionHandler { _, throwable ->
+        _uiState.value = Resource.Error(throwable)
+    }
+
+    private val _uiState = MutableStateFlow(Resource<Unit>())
+    val uiState = _uiState.asStateFlow()
+
+    companion object {
+
+        private const val MAX_RETRIES = 10
+
+        suspend fun generateRecommendedReadingList() {
```

Comment:
Would it make sense to put all of this logic into the `RecommendedReadingListTask` class instead? (Does the ViewModel need to know about any of this?)

Reply:
We will need to generate a new list when the user completes the onboarding process.